Libraries/PeriphDrivers/Source/AFE/afe.c

```diff
@@ -86,30 +86,41 @@
 #define AFE_SPI_PORT MXC_SPI1
 #endif
 
-#define AFE_SPI_BAUD 100000 // Can only run up to PCLK speed
+#define AFE_SPI_BAUD 1000000 // Can only run up to PCLK speed
 #define AFE_SPI_BIT_WIDTH 8
 #define AFE_SPI_SSEL_PIN 1
 #define AFE_SPI_ADDRESS_LEN 1
 
 // AFE Trim Storage Defines
 //#define DUMP_TRIM_DATA
 
+//
+// Enabling this will use the afe timer to timeout polling of the AFE.
+//  This avoid any chance of the blocking function afe_spi_transceive from never returning.
+//  However, as this low level function can be called several times for even a single
+//  AFE register read, it decreases throughput to and from the AFE.
+//  When running at lower system clock frequencies this can be more of a concern.
+//
+//#define AFE_SPI_TRANSCEIVE_SAFE_BUT_SLOWER
+
+#ifdef AFE_SPI_TRANSCEIVE_SAFE_BUT_SLOWER
 //
 // This timeout prevents infinite loop if AFE is not responsive via SPI
 //
 //  Each read from AFE is 8 bits, unless CRC is enabled.
 //  maximum of 4 reads.
 //      1/AFE_SPI_BAUD * 4 * 8 should be a reasonable timeout for this.
-//      1/100000 = 10us * 4 * 8 = 320us
+//      1/1000000 = 1us * 4 * 8 = 32us
 //
-//  Since this is just for catastrophic lockup, lets round it up to 1000
+//  Since this is just for catastrophic lockup, lets round it up to 100
 //      this will also be a even multiple for MXC_AFE_SPI_RESET_TIMEOUT_ITERATES
 //
-#if (AFE_SPI_BAUD != 100000)
+#if (AFE_SPI_BAUD != 1000000)
 #warning "Recalculate MXC_AFE_SPI_READ_TIMEOUT, since baud rate was modified.\n"
```

Comment:
In theory you could calculate the read timeout and then warn if the calculation gave a too small or big of a number.

Reply:
I'll keep this in mind for the future, but current I expect this portion of code will not be used, as it adds too much overhead.  By default, AFE_SPI_TRANSCEIVE_SAFE_BUT_SLOWER is disabled.